src/Foundation/Console/KeyGenerateCommand.php

```diff
@@ -3,80 +3,50 @@
 use Illuminate\Filesystem\Filesystem;
 use Illuminate\Foundation\Console\KeyGenerateCommand as KeyGenerateCommandBase;
 
+use Winter\Storm\Parse\EnvFile;
+
 class KeyGenerateCommand extends KeyGenerateCommandBase
 {
     /**
-     * Create a new key generator command.
+     * Write a new environment file with the given key.
      *
-     * @param  \Illuminate\Filesystem\Filesystem  $files
+     * @param  string  $key
      * @return void
      */
-    public function __construct(Filesystem $files)
+    protected function writeNewEnvironmentFileWith($key)
     {
-        parent::__construct();
+        $currentKey = $this->laravel['config']['app.key'];
 
-        $this->files = $files;
+        $env = EnvFile::open($this->laravel->environmentFilePath());
+        $env->set('APP_KEY', $key);
+        $env->write();
     }
 
     /**
-     * Execute the console command.
+     * Confirm before proceeding with the action.
      *
-     * @return void
-     */
-    public function handle()
-    {
-        $key = $this->generateRandomKey();
-
-        if ($this->option('show')) {
-            return $this->line('<comment>'.$key.'</comment>');
-        }
-
-        // Next, we will replace the application key in the config file so it is
-        // automatically setup for this developer. This key gets generated using a
-        // secure random byte generator and is later base64 encoded for storage.
-        if (!$this->setKeyInConfigFile($key)) {
-            return;
-        }
-
-        $this->laravel['config']['app.key'] = $key;
-
-        $this->info("Application key [$key] set successfully.");
-    }
-
-    /**
-     * Set the application key in the config file.
+     * This method only asks for confirmation in production.
      *
-     * @param  string  $key
+     * @param  string  $warning
+     * @param  \Closure|bool|null  $callback
      * @return bool
      */
-    protected function setKeyInConfigFile($key)
+    public function confirmToProceed($warning = 'Application In Production!', $callback = null)
```

Comment:
@jaxwilko does this confirmation implementation match the changes @LukeTowers made to command confirmations in the 1.2 branch recently?

Reply:
This command extends the Laravel base command, not Winter's. This is only implemented here to override the default message shown to the user when the command is run and a key exists. The confirmation behaviour I recently added is available through a trait, but we don't require confirmation with input in this case, just yes / no.